vcf-dumper/vcf-dumper-ws/src/main/java/uk/ac/ebi/eva/vcfdump/server/HtsgetVcfController.java

```diff
@@ -0,0 +1,235 @@
+            throws IllegalAccessException, IllegalOpenCGACredentialsException, InstantiationException, IOException,
+            StorageManagerException, URISyntaxException, ClassNotFoundException {
+
+        String dbName = "eva_" + species;
+
+        MultivaluedMap<String, String> queryParameters = new MultivaluedHashMap<>();
+        queryParameters.put(VariantDBAdaptor.REGION, Collections
+                .singletonList(chrRegion));
+
+        StreamingResponseBody responseBody = getStreamingBlockResponse(dbName, studies, evaProperties,
+                                                                       queryParameters, response);
+        return responseBody;
+    }
+
+
+    private StreamingResponseBody getStreamingHeaderResponse(String dbName, List<String> studies,
+                                                             Properties evaProperties,
+                                                             MultivaluedMap<String, String> queryParameters,
+                                                             HttpServletResponse response) {
+        return outputStream -> {
+            VariantExporterController controller;
+            try {
+                controller = new VariantExporterController(dbName, studies, outputStream, evaProperties,
+                                                           queryParameters);
+                // tell the client that the file is an attachment, so it will download it instead of showing it
+                response.addHeader(HttpHeaders.CONTENT_DISPOSITION,
+                                   "attachment;filename=" + controller.getOutputFileName());
+                controller.writeHeader();
+            } catch (Exception e) {
+                throw new WebApplicationException(e);
+            }
+        };
+    }
+
+    private StreamingResponseBody getStreamingBlockResponse(String dbName, List<String> studies,
+                                                            Properties evaProperties,
+                                                            MultivaluedMap<String, String> queryParameters,
+                                                            HttpServletResponse response) {
+        return outputStream -> {
+            VariantExporterController controller;
+            try {
+                controller = new VariantExporterController(dbName, studies, outputStream, evaProperties,
+                                                           queryParameters);
+                // tell the client that the file is an attachment, so it will download it instead of showing it
+                response.addHeader(HttpHeaders.CONTENT_DISPOSITION,
+                                   "attachment;filename=" + controller.getOutputFileName());
+                controller.writeBlock();
+            } catch (Exception e) {
+                throw new WebApplicationException(e);
+            }
+        };
+    }
+
+    private class HtsGetError {
```

Comment:
Making this class inner and private reduces the ability to test cases of error: some libraries allow to convert JSON responses to objects directly instead of manually parsing it, so it is useful to have access to the type of the response.

If the idea is not to make visible to other packages, it could be a package protected class like `HtsGetResponse`.